tests/test_ilastik/test_workflows/test_HeadlessPixelClassificationWorkflow.py

```diff
@@ -0,0 +1,178 @@
+from pathlib import Path
+import shutil
+import subprocess
+import logging
+import tempfile
+import json
+import os
+
+import pytest
+import numpy
+import vigra
+import h5py
+import z5py
+import zipfile
+from ndstructs import Array5D, Shape5D
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+
+pytest_plugins = ["pytester"]
+
+try:
+    from lazyflow.distributed.TaskOrchestrator import TaskOrchestrator
+
+    MPI_DEPENDENCIES_MET = bool(shutil.which("mpiexec"))
+except ImportError:
+    MPI_DEPENDENCIES_MET = False
+
+
+@pytest.fixture
+def sample_projects_dir(tmp_path: Path) -> Path:
+    test_data_path = Path(__file__).parent.parent / "data"
+    sample_projects_zip_path = test_data_path / "test_projects.zip"
+    sample_data_dir_path = test_data_path / "inputdata"
+
+    projects_archive = zipfile.ZipFile(sample_projects_zip_path, mode="r")
+    projects_archive.extractall(path=tmp_path)
+
+    shutil.copytree(sample_data_dir_path, tmp_path / "inputdata")
+
+    return tmp_path
+
+
+@pytest.fixture
+def pixel_classification_ilp_2d3c(sample_projects_dir: Path) -> Path:
+    return sample_projects_dir / "PixelClassification2d3c.ilp"
+
+
+def create_h5(data: numpy.ndarray, axiskeys: str) -> Path:
+    assert len(axiskeys) == len(data.shape)
+    path = tempfile.mkstemp()[1] + ".h5"
+    with h5py.File(path, "w") as f:
+        ds = f.create_dataset("data", data=data)
+        ds.attrs["axistags"] = vigra.defaultAxistags(axiskeys).toJSON()
+
+    return Path(path) / "data"
+
+
+class FailedHeadlessExecutionException(Exception):
+    pass
+
+
+def run_headless_pixel_classification(
+    testdir,
+    *,
+    num_distributed_workers: int = 0,
+    project: Path,
+    raw_data: Path,
+    output_filename_format: str,
+    input_axes: str = "",
+    output_format: str = "hdf5",
+    ignore_training_axistags: bool = False,
+):
+    assert project.exists()
+    assert raw_data.parent.exists()
+
+    ilastik_dot_py = Path(__file__).parent.parent.parent.parent / "ilastik.py"
+    subprocess_args = [
+        "python",
+        str(ilastik_dot_py),
+        "--headless",
+        "--project=" + str(project),
+        "--raw-data=" + str(raw_data),
```

Comment:
interesting, wouldn't have thought this works with kebap-case for raw data.
```suggestion
        "--raw_data=" + str(raw_data),
```

Reply:
As I was reworking the cmd line parsing, I added this to `dataSelectionApplet.py`:
```py3
    arg_parser.add_argument(
        "--" + arg_name,
        "--" + arg_name.replace("_", "-"), #allow kebap-case
        nargs="+",
        help=f"List of input files for the {role_name} role",
    )
```

This allows for both cases and prevents some silly errors we would get before, where ilastik could think that `--raw-data` was the name of a file

Reply:
:+1:, I like that!